### PR TITLE
feat: add organization(s) into ecosystem

### PR DESCRIPTION
**Pull request by bhavanakarwade**
### What?
- Developed a POST API enabling the addition of organizations to the Ecosystem
- Modified GET API for fetching all organizations, now capable of filtering organizations by role

### Why?
- By implementing the POST API, ecosystem lead can now seamlessly add organizations to the Ecosystem, enhancing its flexibility and scalability.
- The modification to the GET API allows users to filter organizations based on specific roles

### CODE REVIEW

File: apps/ecosystem/src/ecosystem.service.ts [hunk 1]
```diff
@@ -419,7 +421,76 @@ export class EcosystemService {
     }
   }
 
-  async checkLedgerMatches(orgDetails: OrganizationData, ecosystemId: string): Promise<boolean> {
+
+async addOrganizationInEcosystem(ecosystemLeadOrgs: IEcosystemLeadOrgs): Promise<{ results: { statusCode: number, message: string, error?: string, data?: { orgId: string } }[], statusCode: number, message: string }> {
```
[the remark this answers is not in the document]
Comment (bhavanakarwade) [reply apps/ecosystem/src/ecosystem.service.ts]: Resolved

File: apps/ecosystem/src/ecosystem.service.ts [hunk 2]
```diff
@@ -419,7 +421,76 @@ export class EcosystemService {
...
+        message: ''
+      };
+
+      if (checkOrgExists) {
+        const orgAgentDetails = await this.ecosystemRepository.getAgentDetails(orgId);
+
+        if (orgAgentDetails?.orgDid) {
+          const existingOrg = await this.ecosystemRepository.checkOrgExistsInEcosystem(orgId, ecosystemLeadOrgs.ecosystemId);
+          if (!existingOrg) {
+            await this.ecosystemRepository.addOrganizationInEcosystem(orgId, ecosystemLeadOrgs.ecosystemId, ecosystemLeadOrgs.userId, ecosystemRoleDetails.id);
```
[the remark this answers is not in the document]
Comment (bhavanakarwade) [reply apps/ecosystem/src/ecosystem.service.ts]: Resolved

File: apps/ecosystem/src/ecosystem.repository.ts [hunk 3]
```diff
@@ -87,6 +87,63 @@ export class EcosystemRepository {
...
+   //eslint-disable-next-line camelcase
+    async addOrganizationInEcosystem(orgs: IEcosystemOrgs[], orgIds: string[], ecosystemId: string): Promise<[Prisma.BatchPayload, IEcosystemOrgsData[]]> {
+    try {
+      const result = await this.prisma.$transaction([
+
+      this.prisma.ecosystem_orgs.createMany({
+        data: orgs
+      }),
+
+      this.prisma.ecosystem_orgs.findMany({
```
Comment (shitrerohit) [inline apps/ecosystem/src/ecosystem.repository.ts:99]: @bhavanakarwade Create separate method for this query for re-usability.

File: apps/ecosystem/src/ecosystem.service.ts [hunk 4]
```diff
@@ -419,7 +422,106 @@ export class EcosystemService {
...
+          result.error = `${ResponseMessages.ecosystem.error.unableToAdd}`;
+          result.data = { orgId };
+          errorCount++;
+        }
+        if (0 !== result.statusCode) {
+          results.push(result);
+        }
+      }
+
+      const orgs = addedOrgs.map((item) => item.orgId);
```
Comment (shitrerohit) [inline apps/ecosystem/src/ecosystem.service.ts:491]: @bhavanakarwade Please add one more check to check the length of `addedOrgs` array length and throw error.

Comment (shitrerohit) [inline apps/ecosystem/src/ecosystem.service.ts:491]: @bhavanakarwade Please add one more check to check the length of `addedOrgs` array length and throw error.

File: apps/ecosystem/src/ecosystem.repository.ts (hunk 3 above)
Comment (bhavanakarwade) [reply apps/ecosystem/src/ecosystem.repository.ts]: Resolved

**Review by KulkarniShashank — APPROVED** [review]
LG

**Review by tipusinghaw — APPROVED** [review]
LGTM

**Pull request merged**